zimui/src/stores/main.ts

```diff
@@ -0,0 +1,39 @@
+import { defineStore } from 'pinia'
+import axios from 'axios'
+export const useMainStore = defineStore('main', {
+  state: () => ({
+    channel_data: null,
+    is_loading: false,
+    error: '',
+  }),
+  getters: {},
+  actions: {
+    async fetchChannel() {
+      this.is_loading = true
+      this.error = ''
+      try {
+        const data = await axios.get('./channel.json')
```

Comment:
Isn't axios returning promises? If so we'd better use the promise syntax.

Reply:
Done